### ISSUE TITLE
Module cause compilation error

### ISSUE THREAD

**Issue by AurelienFT** [vlang/v#456]
When I use module i have compilation error I put you an example:
`main.v`:
```
module main

import a

fn main() {
    a.test()
}
```
`a/a.v`:
```
module a

fn test() {
    println("test")
}
```
When I compile with : ` v main.v -lib $PWD/a`
I got the following error:
```
Building module a...
Cross compiling /home/afoucault//.vlang0.0.12//vlib//a.o
V panic: "/home/afoucault//.vlang0.0.12//vlib/builtin/array.vh" doesn't exist
```

**Comment by fabulousduck** [comment]
seems to have been fixed by @Henrixounez

**Comment by AtjonTV** [comment]
For me this issue is a bit more complex as it seems.

I have made a small test app which should use a self made vector module to make vector calculations.

It does not compile as it should.

I want to make this as easy to debug as possible so here is the [complete source code and error tracking](https://gitlab.atvg-studios.at/atjontv/testing-v/tree/master/vector_tests) from my side.

**Comment by AtjonTV** [comment]
**Question**:

Can this be fixed by allowing the use of compile ordering?

**Example**:

You have a folder called `vector` that contains two files.  
You want those two (which use `module vector`) to be a "library" of your application.

You also have a `main.v` file that contains the entry point.

In order to compile it, you would say `v vector/ main.v`.  
Here V would first compile the `vector/` content and make a pseudo library out of it.  
Then it would compile the `main.v` content and link the pseudo library to it so in `main.v` you can say `import vector`.

-----

Well, or something like this.

A similar solution is just to look for a folder with the `import` name where the compiler runs, and do the same without the user telling the compiler to do so. (When the folder is not found, look for it in VROOT as it does now)

**Comment by AtjonTV** [comment]
When you recompile from git, this issue should be fixed :D

**Comment by crose1988** [comment]
> When you recompile from git, this issue should be fixed :D

@AtjonTV The issue is not fixed. When building the [module example in vlang docment](https://vlang.io/docs#modules), i still get a panic: 

```
Building module mymodule...
V panic: "/home/crose//.vlang//vlib/builtin/array.vh" doesn't exist
```

I noticed there was a line of code in you pull request which skip looking for '.vh' file if it's not in builtin directory, that line was not merged into master branch.

I also noticed a comment in main branch:
```
// If we are in default mode, we don't parse vlib .v files, but header .vh files in
// TmpPath/vlib
// These were generated by vfmt
```
If I undestand correctly, when using '-lib', we are in build mode  rather than default mode, thus the complier should not looking for '.vh' files, maybe that's the cause of the panic problem? And what is the vfmt tool?

Thanks and sorry for my poor english :)
